### PR TITLE
14-getArticleById

### PR DESCRIPTION
**Pull request by adamgrime**

### CODE REVIEW

**Review by dav3rid — COMMENTED** [review]
Looking good Adam!
My only issue here is with the test for the happy path on this endpoint.
As we are requesting a specific resource (article 11 in this case) we want to test for its specific properties.
For example, if you responded with article 5, would this test still pass?

You can still use `expect.objectContaining` to give yourself wiggle room for additional properties in the future, but I wouldn't use `expect.any` in this case. Apart from `created_at` - expecting it to be a string will be sufficient!

File: __tests__/app.test.js [hunk 1]
```diff
@@ -39,3 +39,40 @@ describe('404 error for invalid URL', () => {
...
+                expect(article).toBeInstanceOf(Object)
+                expect.objectContaining({
+                     article_id: expect.any(Number),
+                    title: expect.any(String),
+                    topic: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number)
+                })
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: what's this doing? you're not making an assertion here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -39,3 +39,40 @@ describe('404 error for invalid URL', () => {
...
+                expect(article).toEqual({
+                     article_id: 11,
+                    title: "Am I a cat?",
+                    topic: "mitch",
+                    author: "icellusedkars",
+                    body: "Having run out of ideas for articles, I am staring at the wall blankly, like a cat. Does this make me a cat?",
+                    created_at: expect.any(String),
+                    votes: 0
+                })
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: brittle test 👀 this is where you want the objectContaining stuff

File: __tests__/app.test.js [hunk 3]
```diff
@@ -39,3 +39,40 @@ describe('404 error for invalid URL', () => {
...
+    });
+    test('should respond with a 404 error if an invalid article ID is passed in', () => {
+        return request(app)
+            .get(`/api/articles/99999`)
+            .expect(404)
+            .then(({ body }) => {
+            expect(body.msg).toBe('No article found for article_id: 99999');
+    });
+    });
+});
```
Comment (daviidbartlett) [inline __tests__/app.test.js:77]: have you tested for an invalid id? eg. `/api/articles/banana`

**Review by daviidbartlett — COMMENTED** [review]
just a few comments!

**Review by emilyb93 — APPROVED** [review]
Good changes

**Pull request merged**